apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.SessionPoolKeyStatistics.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Spanner.Common.V1;
+
+namespace Google.Cloud.Spanner.V1;
+public partial class SessionPool
+{
+    /// <summary>
+    /// A snapshot of statistics for a <see cref="SessionPoolKey"/> within a <see cref="SessionPool"/>.
+    /// </summary>
+    public sealed class SessionPoolKeyStatistics

```

Comment:
This makes it sound like it's statistics about the key itself. I would suggest SessionPoolPartitionStatistics, but as Amanda has said, partition has a particular meaning in Spanner.

If we can find an appropriate name, we might also want to rename SessionPoolKey similarly, to link the two by name - e.g. "SessionPoolSubdivisionKey" and "SessionPoolSubdivisionStatistics". (That's just an example, not a concrete suggestion.)

We should brainstorm the name further.

Reply:
(If we rename SessionPoolKey that has a bunch of follow-on changes, of course.)

Reply:
SessionPoolSegmentStatistics / SessionPoolSegmentKey would work.

Reply:
Hey @jskeet & @amanda-tarafa ,
Will it be fine to use SessionPoolSegmentStatistics / SessionPoolSegmentKey.

Reply:
I've just checked with Amanda - we're both happy with that. Apologies for the extra work :( I'm happy to re-review once you've renamed, to check that we haven't missed anything. (It'll be very easy to miss names *based* on the previous ones.)